nng-sys/build.rs

```diff
@@ -96,63 +402,52 @@ fn build_bindgen() {
             non_exhaustive: true,
         })
         .constified_enum("nng_flag_enum")
-        // NNG_ESYSERR and NNG_ETRANERR are used like flag
+        // NNG_ESYSERR and NNG_ETRANERR are used like flags
         .constified_enum("nng_errno_enum")
         .constified_enum("nng_pipe_ev")
         .use_core()
         .parse_callbacks(Box::new(BindgenCallbacks::default()))
         .size_t_is_usize(true)
-        // Layout tests are non-portable; 64-bit tests are "wrong" size on 32-bit and always fail.
-        // Don't output tests if we're regenerating `src/bindings.rs` (shared by all platforms when bindgen not used)
+        // Layout tests are non-portable; 64-bit tests fail on 32-bit
+        // Don't output tests if we're regenerating `src/bindings.rs`
         .layout_tests(!cfg!(feature = "source-update-bindings"));
 
-    // Ensure that the headers from the vendored nng sources are used
-    // when the `build-nng` feature builds and links the nng library.
-    // If the `build-nng` feature is not set - do *not* add the directory
-    // to the search path in order to make clang find headers "somewhere".
-    if cfg!(feature = "build-nng") {
+    // Add include path for nng headers
+    // When vendored: use nng/include/
+    // When system: bindgen needs to find headers somewhere (either system paths or we'd need to pass NNG_INCLUDE_DIR)
+    if cfg!(feature = "vendored") {
         builder = builder.clang_arg("-Inng/include/");
+    } else if let Some(inc_dir) = get_env_var("NNG_INCLUDE_DIR") {
+        builder = builder.clang_arg(format!("-I{}", inc_dir));
     }
+    // If neither, bindgen will search system paths
 
-    if cfg!(feature = "nng-compat") {
+    if cfg!(feature = "compat") {
         builder = builder.header("src/compat.h");
     }
-    if cfg!(feature = "nng-supplemental") {
+    if cfg!(feature = "supplemental") {
         builder = builder.header("src/supplemental.h");
     }
-    if cfg!(feature = "no_std") {
-        // no_std support
-        // https://rust-embedded.github.io/book/interoperability/c-with-rust.html#automatically-generating-the-interface
-        builder = builder.ctypes_prefix("cty")
-    }
 
-    const BINDINGS_RS: &str = "bindings.rs";
-    let out_file = PathBuf::from(env::var("OUT_DIR").unwrap()).join(BINDINGS_RS);
+    let out_file = PathBuf::from(env::var("OUT_DIR").unwrap()).join("bindings.rs");
     builder
         .generate()
         .expect("Unable to generate bindings")
-        .write_to_file(out_file.to_owned())
+        .write_to_file(&out_file)
         .expect("Couldn't write bindings");
 
     #[cfg(feature = "source-update-bindings")]
     {
         let bindings = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap())
             .join("src")
-            .join(BINDINGS_RS);
-        std::fs::copy(out_file, bindings).expect("Unable to update bindings");
+            .join("bindings.rs");
+        std::fs::copy(&out_file, bindings).expect("Unable to update bindings");
```

Comment:
nit: first char of expect message lowercase to match the panics above. Same for the (not touched) ones above.